scanpy/testing/_pytest/__init__.py

```diff
@@ -79,3 +79,18 @@ def _modify_doctests(request: pytest.FixtureRequest) -> None:
     skip_reason: str | None
     if skip_reason := getattr(func, "_doctest_skip_reason", None):
         pytest.skip(reason=skip_reason)
+
+
+def pytest_itemcollected(item: pytest.Item) -> None:
+    # Dask AnnData tests require anndata > 0.10
+    import anndata
+    from packaging.version import Version
+
+    requires_anndata_dask_support = (
+        len([mark for mark in item.iter_markers(name="anndata_dask_support")]) > 0
+    )
+
+    if requires_anndata_dask_support and Version(anndata.__version__) < Version("0.10"):
+        item.add_marker(
+            pytest.mark.skip(reason="dask support requires anndata version > 0.10")
+        )
```

Comment:
I’m pretty sure this doesn’t have any effect in Pytest 8.0.

I moved to a `autouse=True` fixture with imperative `pytest.skip(reason=...)` in the Pytest 8.0 PR

Reply:
If I comment it out, tests fail. With it, those tests are skipped.

Reply:
OK, so `pytest.mark.usefixtures` support for doctests was removed, see https://github.com/pytest-dev/pytest/issues/11759#issuecomment-1874379413

I’m unsure which marks (if any) are supported when added in `pytest_itemcollected`. I’m not super concerned about this: Maybe we should clear this up, maybe we just do it and if they change behavior, we deal with it.

Reply:
Do you remember why we are using `pytest_itemcollected`? I think I introduced it in anndata for gpu stuff, but also thought that it was based on pytest documentation.

Reply:
The relevant section is called “Reporting hooks”, so maybe they aren’t intended for mutation.